awx/main/tests/functional/api/test_instance_group.py

```diff
@@ -78,6 +78,20 @@ def instance_group_jobs_successful(instance_group, create_job_factory, create_pr
     return jobs_successful + project_updates_successful
 
 
+@pytest.mark.django_db
+def test_instance_group_is_isolated(instance_group, isolated_instance_group):
+    # `isolated_instance_group` is isolated, `instance_group` is not
+    assert not instance_group.is_isolated
+    assert isolated_instance_group.is_isolated
+
+
+@pytest.mark.django_db
+def test_instance_group_is_controller(instance_group, isolated_instance_group):
+    # `instance_group` controls `isolated_instance_group`, hence a controller
+    assert instance_group.is_controller
+    assert not isolated_instance_group.is_controller
+
+
```

Comment:
Maybe add a sanity check to make sure that empty instance groups aren't considered an isolated group or controller group?

Reply:
What I wrote wasn't very clear. Basically, was thinking it'd be worth adding a test for the special case where an instance group is empty. If it was an isolated group (i.e. has a controller group) before it lost all of its nodes, it should still be an isolated group. Similarly, if it was just a regular instance group before all nodes were removed, then it should still be a regular instance group.